hal/common/file.hpp

```diff
@@ -65,7 +65,7 @@ class File
 	 * 
 	 * \return  success
 	 */
-	virtual bool open(const char* path) = 0;
+	virtual bool open(const char* path, bool new_file) = 0;
```

Comment:
Unfortunately yes. 
It is needed to create a log file with the ID of the platform. The ID of the platform is set (latest) after having read the flash, therefore, we can create/open a file with the correct name only after reading the flash. This is too late for the creator of the class object file... 
Therefore, by setting the new_file to false, you create the file only after having read the flash. 

If you have another idea on how to do this, please let me know